server/src-lib/Hasura/GraphQL/Schema/Remote.hs

```diff
@@ -0,0 +1,424 @@
+module Hasura.GraphQL.Schema.Remote
+  ( buildRemoteParser
+  , remoteFieldFullSchema
+  , inputValueDefinitionParser
+  , lookupObject
+  , lookupType
+  , lookupScalar
+  ) where
+
+import           Hasura.Prelude
+import           Hasura.RQL.Types
+import           Hasura.SQL.Types
+
+import           Language.GraphQL.Draft.Syntax       as G
+import qualified Data.List.NonEmpty                  as NE
+import           Data.Type.Equality
+
+import           Hasura.GraphQL.Parser               as P
+import qualified Hasura.GraphQL.Parser.Internal.Parser as P
+
+buildRemoteParser
+  :: forall m n
+   . (MonadSchema n m, MonadError QErr m)
+  => IntrospectionResult
+  -> RemoteSchemaInfo
+  -> m ( [P.FieldParser n (RemoteSchemaInfo, Field NoFragments Variable)]
+       , Maybe [P.FieldParser n (RemoteSchemaInfo, Field NoFragments Variable)]
+       , Maybe [P.FieldParser n (RemoteSchemaInfo, Field NoFragments Variable)])
```

Comment:
I think there's a type alias for this: `RQL.Types.SchemaCache.ParsedIntrospection`?

Reply:
Not quite - `ParsedIntrospection` isn't polymorphic.

Reply:
Gotcha. Maybe it should be though? 'cause it feels very similar. But perhaps we can harmonize that later.